src/core/Table/Table.tsx

```diff
@@ -118,6 +118,11 @@ export type TableProps<
    * Content shown when there is no data after filtering.
    */
   emptyFilteredTableContent?: React.ReactNode;
+  /**
+   * Function that returns whether row is disabled or not.
+   * It also disables selection and expansion cells.
+   */
+  isRowDisabled?: (rowData: T) => boolean;
```

Comment:
This prop is a bit unintuitive. Wouldn't it be better to have `isDisabled` property for each row in `data`? You can access it from `row.original`.

Reply:
But it is easier to use this way because I really doubt that there is a project that gets data from the server with the field `isDisabled` 😃 also if there is you can just pass `rowData => rowData.isDisabled` or `rowData => rowData.status !== 'success`. Another thing that users won't need to remap their data to have extra field for disabled state.

Reply:
> But it is easier to use this way

"easier" is subjective. I just thought it would make sense to have `isDisabled`, similar to `isExpanded` and `isSelected` properties of row. But those are not passed by the user, so maybe it's fine this way 😕 🤷‍♀️ 👍

Reply:
To alleviate the unintuitiveness, we could improve the JSDoc comment. Instead of _`Function that returns whether row is disabled or not. It also disables selection and expansion cells.`_, maybe we should be more clear about what is expected. Something like: _`Should return true if a row is disabled (i.e. cannot be selected or expanded). If not specified, all rows are enabled.`_.

Another source of confusion can be the param type (someone might ask "What is `rowData: T`?"). We could give an example to clarify that.

Reply:
Changed description. About `rowData`, we have it in several places and IDE should show your data type instead of `T` so I don't think it is worth to put example on every prop, we have Storybook for that.